packages/block-editor/src/components/inserter/index.js

```diff
@@ -370,10 +371,17 @@ export default compose( [
 					blockToInsert = createBlock( allowedBlockType.name );
 				}
 
-				insertBlock( blockToInsert, getInsertionIndex(), rootClientId );
+				insertBlock(
+					blockToInsert,
+					getInsertionIndex(),
+					rootClientId,
+					selectBlockOnInsert
```

Comment:
If `undefined` this will default to `true` meaning that the blocks will continue to be focused in other contexts.